kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/InBandKeyManagerTest.java

```diff
@@ -105,6 +114,229 @@ void shouldEncryptRecordValue() {
                 .isEqualTo(value);
     }
 
+    @Test
+    void shouldPreserveMultipleBatches() {
```

Comment:
this is quite hard to digest, can we refactor to make it more readily understandable?

Reply:
will do, I'll see if I can pull apart the encrypt/decrypt side